pandas/core/index.py

```diff
@@ -826,10 +826,44 @@ def _validate_index_level(self, level):
             raise KeyError('Level %s must be same as name (%s)'
                            % (level, self.name))
 
-    def _get_level_number(self, level):
+    def _get_level_number(self, level, ignore_names=False):
+        """
+        Returns level number corresponding to level.
+        If level is a level name and ignore_names is False,
```

Comment:
can you indent the 2nd parts of lines, prob not PEP8 but easier to read the separate items.